db/models/reviews.model.js

```diff
@@ -4,18 +4,69 @@ exports.fetchReview = (requestQuery) => {
-    .then((reviews) => {
-        return reviews.rows
-    })
+exports.fetchAllReviews = (requestQuery) => {
+
+    if(Object.keys(requestQuery).length === 0){
+        return db.query('SELECT owner, title, reviews.review_id, category, review_img_url, reviews.created_at, reviews.votes, designer, COUNT(comments.review_id) AS comment_count FROM reviews LEFT JOIN comments ON reviews.review_id = comments.review_id GROUP BY reviews.review_id ORDER BY reviews.created_at DESC;')
+        .then((reviews) => {
+            return reviews.rows
+        }) 
+    }
+
+    if(requestQuery.category){
+        let order = "DESC"
+
+        if(requestQuery.order){
+            if(!["asc", "desc"].includes(requestQuery.order)){
+                return Promise.reject({status: "400", msg: "Bad request."})
+            }
+            order = requestQuery.order
+        }
+
+        return db.query(`SELECT owner, title, reviews.review_id, category, review_img_url, reviews.created_at, reviews.votes, designer, COUNT(comments.review_id) AS comment_count FROM reviews LEFT JOIN comments ON reviews.review_id = comments.review_id WHERE category = $1 GROUP BY reviews.review_id ORDER BY reviews.created_at ${order};`, [requestQuery.category])
+        .then((reviews) => {
+            if(reviews.rows.length === 0){
+                return Promise.reject({status: "400", msg: "Bad request."})
+            }
+            return reviews.rows
+        })
+    }
+
+    if(requestQuery.sort_by){
+        let order = "DESC"
+
+        if(!["owner", "title", "review_id", "category", "review_img_url", "created_at", "votes", "designer", "review_id"].includes(requestQuery.sort_by)){
+            return Promise.reject({status: "400", msg: "Bad request."})
+        }
+
+        if(requestQuery.order){
+            if(!["asc", "desc"].includes(requestQuery.order)){
+                return Promise.reject({status: "400", msg: "Bad request."})
+            }
+            order = requestQuery.order
+        }
+        
+        return db.query(`SELECT owner, title, reviews.review_id, category, review_img_url, reviews.created_at, reviews.votes, designer, COUNT(comments.review_id) AS comment_count FROM reviews LEFT JOIN comments ON reviews.review_id = comments.review_id GROUP BY reviews.review_id ORDER BY ${requestQuery.sort_by} ${order};`)
+        .then((reviews) => {
+            return reviews.rows
+        })
+    }
+
+    if(requestQuery.order && Object.keys(requestQuery).length === 1){
+        let order = requestQuery.order
+        
+        return db.query(`SELECT owner, title, reviews.review_id, category, review_img_url, reviews.created_at, reviews.votes, designer, COUNT(comments.review_id) AS comment_count FROM reviews LEFT JOIN comments ON reviews.review_id = comments.review_id GROUP BY reviews.review_id ORDER BY reviews.created_at ${order};`)
+        .then((reviews) => {
+            return reviews.rows
+        })
```

Comment:
Please try to refactor the model to build up the query string dynamically according to the presence of the query. There shouldn't be multiple complete query strings within the code. We want to keep it 'DRY' ('Don't Repeat Yourself') .